Tests/EndToEnd.tests.ts

```diff
@@ -53,7 +58,7 @@ class fakeResponse {
     }
 
     public pass(test = false): void {
-        this.callbacks["data"] ? this.callbacks["data"]("data") : null;
+        this.callbacks["data"] ? this.callbacks["data"](this._responseData) : null;
```

Comment:
Modified this in order to have response return a JSON object as response data, other than string "data".